src/services/CurrencyExchangeRateService.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  RateSourceResponse,
+  RateSourceService,
+} from './rateSource/RateSourceService';
+import rateSourceInstances from './rateSource';
+import responseMessages from '../constants/responseMessages';
+import logResponse from '../utils/logResponse';
+
+export class CurrencyExchangeRateService {
+  constructor(private readonly rateSourceService?: RateSourceService) {}
+
+  public async getRate(): Promise<RateSourceResponse> {
+    if (this.rateSourceService) {
+      return this.rateSourceService.retrieve();
+    }
+
+    for (const source of rateSourceInstances) {
+      try {
+        const response = await source.retrieve();
+        logResponse({ source: source.getUrl, response });
```

Comment:
Please add more info to logs
Like who did what and what happened :)

Reply:
Well, I have 'who' - source of exchange rate, 'what happened' - status code and possible response\error